spydrnet_physical/util/tests/test_ConnectPointList.py

```diff
@@ -0,0 +1,219 @@
+		self.assertEqual(point.connection,(3,3,3,4))
+
+	def test_search_to_point(self):
+		'''This test checks whether the search to point correctly returns connection coming into the given coordinate'''
+		self.pts.cursor = (2,3)
+		self.pts.move_x(1)
+		self.pts.move_y(1)
+		point = self.pts.search_to_point((3,3))
+		self.assertEqual(point.connection,(2,3,3,3))
+
+	def test_push_connection_down(self):
+		'''This test checks whether the level of the connection is going one down'''
+		pt = ConnectPoint(2, 2, 2, 3)
+		self.pts.push_connection_down (pt)
+		self.assertEqual(pt.level,"down")
+
+	def test_pull_connection_up(self):
+		'''This test checks whether the level of the connection is going one down'''
+		pt = ConnectPoint(2, 2, 2, 3)
+		self.pts.pull_connection_up(pt)
+		self.assertEqual(pt.level,"up")
+
+	def test_make_top_connection(self):
+		'''This test checks whether the connection is made with the top level'''
+		pass
+		#pts = ConnectPointList(5,10)
+		#pt = ConnectPoint(2, 2, 2, 3)
+		#pts.make_top_connection()
+		#self.assertEqual(pt.level,"same")
+
+	def test_reset_level(self):
+		'''This test checks whether the connection reset to the same level'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		self.pts.reset_level(pt)
+		self.assertEqual(pt.level,"same")
+
+	def test_hold_cursor(self):
+		'''This test checks whether the cursor holds its position when a connection is made in any coordonate'''
+		self.pts.cursor = (2,3)
+		self.pts.hold_cursor()
+		self.pts.move_x(3)
+		self.assertEqual(self.pts.cursor,(2,3))
+
+	def test_release_cursor(self):
+		'''This test checks whether the cursor also moves its position when a connection is made in any coordonate'''
+		self.pts.set_cursor(2,3)
+		self.pts.hold_cursor()
+		self.pts.move_x(3)
+		self.pts.release_cursor()
+		self.pts.move_x(2).move_y(4)
+		self.assertEqual(self.pts.cursor,(4,7))
+
+	def test_flip_V(self):
+		'''This test checks whether the connection is correctly flipped vertically'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_from_point((2,2))
+		self.assertEqual(point.connection,(2,2,2,3))
+		pts.flip("V")
+		self.assertEqual(point.connection,(2,-2,2,-3))
+
+	def test_flip_H(self):
+		'''This test checks whether the connection is correctly flipped horizontally'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_from_point((2,2))
+		self.assertEqual(point.connection,(2,2,2,3))
+		pts.flip("H")
+		self.assertEqual(point.connection,(-2,2,-2,3))
+
+	def test_sample_connections(self):
+		'''This test checks whether the connection is sampled correctly equal to the max distance defined '''
+		pt = ConnectPoint(2, 2, 5, 2,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_to_point((5,2))
+		self.assertEqual(point.connection,(2,2,5,2))
+		pts.sample_connections(max_distance=1)
+		self.assertEqual(point.connection,(4,2,5,2))
```

Comment:
Check if all three segments exist after sampling, also the level property on that is not changed

Reply:
So here all other segments do exist, but if  the level was going "up" in a ConnectPoint only the last sampled connection would be going "up" while all other segments would remain at "same" level.
for example in this case, sampled connections going from 2,2 to 3,2 and 3,2 to 4,2 would be at "same" level while the connection going from 4,2 to 5,2 would be going "up" a level.
Please confirm this functionality is correct

Reply:
Can you please comment on it.

Reply:
Same -> Same Same Same
Up -> Same Same Up